pkg/registry/apps/statefulset/storage/storage.go

```diff
@@ -92,3 +115,88 @@ var _ rest.ShortNamesProvider = &REST{}
 func (r *REST) ShortNames() []string {
 	return []string{"sts"}
 }
+
+type ScaleREST struct {
+	registry statefulset.Registry
+}
+
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &extensions.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx genericapirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	ss, err := r.registry.GetStatefulSet(ctx, name, options)
+	if err != nil {
+		return nil, errors.NewNotFound(extensions.Resource("statefulsets/scale"), name)
+	}
+	scale, err := scaleFromStatefulSet(ss)
+	if err != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("%v", err))
+	}
+	return scale, err
+}
+
+func (r *ScaleREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
+	ss, err := r.registry.GetStatefulSet(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, errors.NewNotFound(extensions.Resource("statefulsets/scale"), name)
```

Comment:
same here, I'd only expect NotFound errors to cause us to return NotFound errors